gcapi/client.py

```diff
@@ -872,27 +921,46 @@ def create_display_sets_from_images(
 
         Returns
         -------
-        The pks of the newly created display sets.S
+        The pks of the newly created display sets.
         """
         res = []
+        interfaces: Dict[str, Dict] = {}
+        for display_set in display_sets:
+            new_interfaces = yield from self._validate_display_set_values(
+                display_set.items(), interfaces
+            )
+            interfaces.update(new_interfaces)
+
         for display_set in display_sets:
             ds = yield from self.__org_api_meta.reader_studies.display_sets.create(
                 reader_study=reader_study
             )
-            for interface, files in display_set.items():
-                try:
-                    yield from self.__org_api_meta.interfaces.detail(
-                        slug=interface
+            values = []
+            for slug, value in display_set.items():
+                interface = interfaces[slug]
+                data = {"interface": slug}
+                super_kind = interface["super_kind"].casefold()
+                if super_kind == "image":
+                    yield from self._upload_image_files(
+                        display_set=ds["pk"], interface=slug, files=value
                     )
-                except ObjectNotFound as e:
-                    raise ValueError(
-                        f"{interface} is not an existing interface. "
-                        f"Please provide one from this list: "
-                        f"https://grand-challenge.org/algorithms/interfaces/"
-                    ) from e
+                elif super_kind == "file":
+                    upload = yield from self._upload_file(value)
+                    data["user_upload"] = upload["api_url"]
+                else:
+                    if (
+                        isinstance(value, list)
+                        and len(value) == 1
+                        and Path(value[0].exists())
+                    ):
+                        upload = yield from self._upload_file(value)
+                        data["user_upload"] = upload["api_url"]
```

Comment:
Is this really necessary? I thought that json interfaces that are not stored in the db are super_kind `file` now?

Reply:
If not, then there might be a few places in the codebase, where we need to check that. I think for archive items it's done like that right now.

Reply:
No, that change never made it past the PR review.

Reply:
Hm, I think it was actually always like that. At the moment super_kind is determined like this: 
```@property
    def super_kind(self):
        if self.saved_in_object_store:
            if self.is_image_kind:
                return InterfaceSuperKindChoices.IMAGE
            else:
                return InterfaceSuperKindChoices.FILE
        else:
            return InterfaceSuperKindChoices.VALUE
```

So that would suggest that json files that are not stored in the db are of super_kind `File`. That's also how I recently implemented the changes here: https://github.com/comic/grand-challenge.org/pull/2644

Reply:
For completeness, saved_in_object_store is determined as follows: 

```
    @property
    def saved_in_object_store(self):
        # files and images should always be saved to S3, others are optional
        return (
            self.is_image_kind
            or self.is_file_kind
            or not self.store_in_database
        )
```

Reply:
So the highlighted part here can go, I would say ;-)

Reply:
Ah sorry, I was thinking about the `is_file_kind` property.